tests/testthat/test-exporters.R

```diff
@@ -361,3 +357,67 @@ test_that("export_as_doc works thanks to tt_to_flextable", {
 
   expect_true(file.exists(doc_file))
 })
+
+test_that("tt_to_flextable does not create different cells when colcounts (or multiple) on different lines", {
+  skip_if_not_installed("flextable")
+  require("flextable", quietly = TRUE)
+
+  lyt <- basic_table(show_colcounts = TRUE) %>%
+    split_rows_by("ARM", label_pos = "topleft") %>%
+    split_rows_by("STRATA1", label_pos = "topleft") %>%
+    split_cols_by("STRATA1", split_fun = keep_split_levels("B"), show_colcounts = TRUE) %>%
+    split_cols_by("SEX", split_fun = keep_split_levels(c("F", "M"))) %>%
+    split_cols_by("COUNTRY", split_fun = keep_split_levels("CHN")) %>%
+    analyze("AGE")
+
+  tbl <- build_table(lyt, ex_adsl)
+  ft1 <- tt_to_flextable(tbl, counts_in_newline = FALSE)
+  ft2 <- tt_to_flextable(tbl, counts_in_newline = TRUE)
+
+  expect_equal(flextable::nrow_part(ft1, "header"), flextable::nrow_part(ft2, "header"))
+})
+
+test_that("check titles bold and html theme", {
+  skip_if_not_installed("flextable")
+  require("flextable", quietly = TRUE)
+
+  lyt <- basic_table(show_colcounts = TRUE) %>%
+    split_rows_by("ARM", label_pos = "topleft") %>%
+    split_rows_by("STRATA1", label_pos = "topleft") %>%
+    split_cols_by("STRATA1", split_fun = keep_split_levels("B"), show_colcounts = TRUE) %>%
+    split_cols_by("SEX", split_fun = keep_split_levels(c("F", "M"))) %>%
+    split_cols_by("COUNTRY", split_fun = keep_split_levels("CHN")) %>%
+    analyze("AGE")
+
+  tbl <- build_table(lyt, ex_adsl)
+  main_title(tbl) <- "Main title"
+  subtitles(tbl) <- c("Some Many", "Subtitles")
+  main_footer(tbl) <- c("Some Footer", "Mehr")
+
+  expect_silent(ft1 <- tt_to_flextable(tbl, theme = theme_html_default(), bold_titles = FALSE))
+  expect_silent(ft1 <- tt_to_flextable(tbl, theme = theme_html_default(), bold_titles = c(2, 3)))
+  expect_error(ft1 <- tt_to_flextable(tbl, theme = theme_html_default(), bold_titles = c(2, 3, 5)))
+})
+
+
+test_that("check pagination", {
+  skip_if_not_installed("flextable")
+  require("flextable", quietly = TRUE)
+
+  lyt <- basic_table(show_colcounts = TRUE) %>%
+    split_rows_by("ARM", label_pos = "topleft", page_by = TRUE) %>%
+    split_rows_by("STRATA1", label_pos = "topleft") %>%
+    split_cols_by("STRATA1", split_fun = keep_split_levels("B"), show_colcounts = TRUE) %>%
+    split_cols_by("SEX", split_fun = keep_split_levels(c("F", "M"))) %>%
+    split_cols_by("COUNTRY", split_fun = keep_split_levels("CHN")) %>%
+    analyze("AGE")
+
+  tbl <- build_table(lyt, ex_adsl)
+
+  main_title(tbl) <- "Main title"
+  subtitles(tbl) <- c("Some Many", "Subtitles")
+  main_footer(tbl) <- c("Some Footer", "Mehr")
+  prov_footer(tbl) <- "Some prov Footer"
+
+  expect_silent(out <- tt_to_flextable(tbl, paginate = TRUE, lpp = 100))
```

Comment:
can we snapshot this?

Reply:
The reason I ask is that I was thinking to write a s3 function of as_flextable.rtables, then we can fully utilize flextable functions such as information_data_chunk, and etc, instead of us writing functions. perhaps next version. perhaps in the next version of this function, we could do something like this.

Reply:
I see extending as_flextable... It may makes sense but we need probably to import it at that time? 
I can snapshot that yes, but only the data, the style is a bit more complex

Reply:
I will work on the pagination precisely in the next PR